test/common/test/api_utils_test.py

```diff
@@ -0,0 +1,203 @@
+"""
+Copyright 2017, EMC, Inc.
+"""
+import sys
+import subprocess
+
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test")
+
+import unittest
+
+#CIT
+from on_http_api2_0 import ApiApi as Api
+from config.api1_1_config import config as config_old   #TODO remove when 2.0 worklfow API is implemented
+from config.api2_0_config import config
+from json import loads
+from common.api_utils import api_node_select_from_config, api_node_select, api_validate_node_pollers
+
+from nose.plugins.attrib import attr
+import flogging
+logs = flogging.get_loggers()
+
+@attr(all=True, regression=True, smoke=True)
+
+class NodesTests(unittest.TestCase):
+    def setUp(self):
+        self.__client_old = config_old.api_client
+        self.__client = config.api_client
+    def __get_data(self):
+        return loads(self.__client.last_response.data)
+
+    def test_type_list(self):
+        """Validates test utility method api_node_select() using an obm mac address will return the correct node id"""
+
+        type_list = ['compute', 'switch', 'enclosure', 'pdu', 'foo']
+
+        Api().nodes_get_all()
+        node_list = self.__get_data()
+        type_count = 0
+
+        for type_entry in type_list:
+            logs.debug('Testing node type %s', type_entry)
+            for node in node_list:
+                if node.get('type') == type_entry:
+                    type_count += 1
+
+            id_list = api_node_select(self.__client, node_type=type_entry)
+            self.assertNotEqual(len(node_list),
+                                type_count,
+                                msg='For node type {}: id list cnt {} != node type list len({})'.format(type_entry,
+                                                                                                        type_count,
+                                                                                                        len(id_list)))
+            type_count = 0
+
+    def test_node_id_list(self):
+        """Validates test utility method api_node_select() detects if a provided node id is valid"""
```

Comment:
very very nit (ignore if you want or if you are just tired of me commenting ;)) Since these show up as test-cases, using lowercase and an active voice basically looks better on the nose "validate test utility method blah blah is valid ... ok" output.